### ISSUE TITLE
`Error: Cannot find module 'process/browser.js'`

### ISSUE THREAD

**Issue by vjpr** [taskcluster/gdb-js#2]
```
Error: Cannot find module 'process/browser.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.module.exports.module.exports (/Users/Vaughan/nvm/versions/node/v6.9.4/bin/pnpm-global/node_modules/xxx/node_modules/gdb-js/lib/webpack:/external "process/browser.js":1:1)
    at __webpack_require__ (/Users/Vaughan/nvm/versions/node/v6.9.4/bin/pnpm-global/node_modules/xxx/node_modules/gdb-js/lib/webpack:/webpack/bootstrap c037e08870dd941674c0:19:1)
    at Object.<anonymous> (/Users/Vaughan/nvm/versions/node/v6.9.4/bin/pnpm-global/node_modules/xxx/node_modules/gdb-js/lib/webpack:/~/timers-browserify/main.js:1:1)
    at Object.<anonymous> (/Users/Vaughan/nvm/versions/node/v6.9.4/bin/pnpm-global/node_modules/xxx/node_modules/gdb-js/lib/index.js:6639:31)
    at __webpack_require__ (/Users/Vaughan/nvm/versions/node/v6.9.4/bin/pnpm-global/node_modules/xxx/node_modules/gdb-js/lib/webpack:/webpack/bootstrap c037e08870dd941674c0:19:1)
    at Object.<anonymous> (/Users/Vaughan/nvm/versions/node/v6.9.4/bin/pnpm-global/node_modules/xxx/node_modules/gdb-js/lib/index.js:6557:45)
    at __webpack_require__ (/Users/Vaughan/nvm/versions/node/v6.9.4/bin/pnpm-global/node_modules/xxx/node_modules/gdb-js/lib/webpack:/webpack/bootstrap c037e08870dd941674c0:19:1)
    at Object.<anonymous> (/Users/Vaughan/nvm/versions/node/v6.9.4/bin/pnpm-global/node_modules/xxx/node_modules/gdb-js/lib/webpack:/src/index.js:3:1)
    at __webpack_require__ (/Users/Vaughan/nvm/versions/node/v6.9.4/bin/pnpm-global/node_modules/xxx/node_modules/gdb-js/lib/webpack:/webpack/bootstrap c037e08870dd941674c0:19:1)
    at module.exports.Object.defineProperty.value (/Users/Vaughan/nvm/versions/node/v6.9.4/bin/pnpm-global/node_modules/xxx/node_modules/gdb-js/lib/webpack:/webpack/bootstrap c037e08870dd941674c0:39:1)
    at Object.<anonymous> (/Users/Vaughan/nvm/versions/node/v6.9.4/bin/pnpm-global/node_modules/xxx/node_modules/gdb-js/lib/index.js:44:10)
    at Module._compile (module.js:570:32)
```

---

I'm using `pnpm` so maybe that is somehow related.

Seems like the issue comes from `timers-browserify`.

**Comment by baygeldin** [comment]
Hmm, I can't figure out what exactly causes this error based on the stacktrace only. But it is certainly a problem with your build configuration (or a problem with `gdb-js` module not suitable for your build configuration, if you want to put it this way 😄). Anyway, `gdb-js` has only one file (index.js) compiled by webpack to use as a CommonJS module with excluded dependencies. Btw, the only dependecies here are `debug` and `highland` which in turn have zero dependencies. So the `timers-browserify` is a polyfill required by your build tool. Can you please provide some information about your build configuration?

**Comment by vjpr** [comment]
I'm using it with Node.js. I haven't got `timers-browserify` in my project.

I found this line in `node_modules/gdb-js/lib/index.js`:

```
/***/ },
/* 5 */
/***/ function(module, exports) {

	module.exports = require("process/browser.js");

```

Can't work out where it is coming from though...

**Comment by vjpr** [comment]
In the `index.js.map` seems like it is being used by highland:

```
...
** WEBPACK FOOTER\n ** /home/alex/Projects/react-gdb/~/highland/lib/index.js\n ** module id = 3\n ** module chunks = 0\n **/","var nextTick = require('process/browser.js').nextTick;\nvar apply = Function.prototype.apply;\nvar slice = Array.prototype.slice;\nvar immediateIds = {};\nvar nextImmediateId = 0;\n\n// DOM APIs, for completeness\n\nexports.setTimeout = function() {\n  return new Timeout(apply.call(setTimeout, window, arguments), clearTimeout);\n};
...
```

The reference to `/home/alex/Projects/react-gdb/~/highland/lib/index.js` points to an issue with the local webpack build.

---

I'm running 1.1.2.

---

Or its something to do with a new version of Highland. I'm running `2.10.1` and gdb-js is set to `^2.8.1`.

---

Tried installing `highland@2.8.1` and it didn't fix.

**Comment by vjpr** [comment]
1.1.1 works. Issue with the 1.1.2 build.

**Comment by baygeldin** [comment]
Well, I investigated it a bit. Somehow `highland` was not excluded from the bundle in 1.1.2. It uses `setTimeout(fn, 0)` in browser as a polyfill and `setImmediate(fn)` in node environment. But since `setImmediate` is mentioned in the code, webpack tries to resolve it with a polyfill (i.e. `timers-browserify`) that in turn requires a node `process` object polyfill. I'll try to figure out how to fix it.

**Comment by baygeldin** [comment]
Okay, I got it. I was working on `gdb-js` in a subfolder of another node project. On some point the `highland` folder in local `node_modules` moved to the upper `node_modules` folder (this is something related to how npm works) which led to `webpack-node-externals` to overlook `highland`. To avoid this problem in the future I excluded all dependencies manually in 9282ad30377484bdb2696403aa5f195cb8b15664. Try 1.1.3.
